### ISSUE TITLE
`seed` function in `Hook` should get AppContext as an argument.

### ISSUE THREAD

**Issue by DenuxPlays** [loco-rs/loco#1060]
## Feature Request

Currently the seed function only gets the `DatabaseConnection` but it should get the `AppContext` as an argument.

**Describe the solution you'd like**

Yeah basically as I said.
Change the arguments or extend it.

**Describe alternatives you've considered**

None

**Extras**

I would like to work on this if it gets approved.

**Comment by DenuxPlays** [comment]
Makes more sense together with: https://github.com/loco-rs/loco/issues/972

**Comment by DenuxPlays** [comment]
> can you describe the uae case?

Considering the other ticket (#972) it might be useful to get the AppContext as an argument.
For example I have services that need AppContext as an argument to create them.
Currently it's useless as only files can get used to create fixtures.

In my case I have a Snowflake Generator which i need to use inside the fixtures but the generator requires the AppContext as an argument.

TLDR:
It's only useful with #972 if this ticket/pr gets declined this is to no use.
